dataplane/internal/kernel/kernel.go

```diff
@@ -24,37 +24,50 @@ import (
 )
 
 // SetInterfaceHWAddr sets the MAC address of a network interface.
-func SetInterfaceHWAddr(name string, addr net.HardwareAddr) error {
+func SetInterfaceHWAddr(name string, addr string) error {
 	link, err := netlink.LinkByName(name)
 	if err != nil {
 		return fmt.Errorf("failed to get interface: %w", err)
 	}
-	if err := netlink.LinkSetHardwareAddr(link, addr); err != nil {
+	addrBytes, err := net.ParseMAC(addr)
+	if err != nil {
+		return fmt.Errorf("failed to set parse addres: %v", err)
+	}
+	if err := netlink.LinkSetHardwareAddr(link, addrBytes); err != nil {
 		return fmt.Errorf("failed to get hwaddr of link: %w", err)
 	}
 	return nil
 }
 
-// SetInterfaceIPs sets the IP addresses of a network interface.
-// Any existing IPs on the interface are deleted.
-func SetInterfaceIPs(name string, ips []*net.IPNet) error {
+// SetInterfaceIP sets the IP addresses of a network interface.
+func SetInterfaceIP(name string, ip string, prefixLen int) error {
 	link, err := netlink.LinkByName(name)
 	if err != nil {
 		return fmt.Errorf("failed to get interface: %w", err)
 	}
-	currIPs, err := netlink.AddrList(link, unix.AF_UNSPEC)
-	if err != nil {
-		return fmt.Errorf("failed to get existing ips: %w", err)
+	ipNet := &net.IPNet{}
+	ipNet.IP = net.ParseIP(ip)
+	if ipNet.IP == nil {
+		return fmt.Errorf("failed to parse ip")
 	}
-	for i := range currIPs {
-		if err := netlink.AddrDel(link, &currIPs[i]); err != nil {
-			return fmt.Errorf("failed to add delete existing IP: %w", err)
-		}
+	ipNet.Mask = net.CIDRMask(prefixLen, 128)
+	if ipNet.IP.To16().To4() != nil {
```

Comment:
what is this pattern? i'm not understanding it.

Reply:
added comment